app/routes/index.tsx

```diff
@@ -1,8 +1,26 @@
+import { json, LoaderArgs } from "@remix-run/node";
+import { useLoaderData } from "@remix-run/react";
+import { requireUsername } from "~/session.server";
+
+export const loader = async ({ request }: LoaderArgs) => {
+  const username = await requireUsername(request);
+
+  return json({ username: username });
+};
+
 export default function Index() {
+  const data = useLoaderData<typeof loader>();
+
   return (
     <main>
       <h1>Starchart</h1>
       <p>Start making your own custom domains and certificates today!</p>
+      <p>{data.username}</p>
```

Comment:
Let's use [Chakra UI components](https://chakra-ui.com/docs/components) and [Remix's `<Form>` component](https://remix.run/docs/en/v1/components/form).  Here's a suggestion:

```tsx
<Text>Welcome {data.username}</Text>
<Form action="/logout" method="post">
  <Button type="submit" colorScheme="red">Logout</button>
</Form>
```